src/helperFunctions/process.py

```diff
@@ -106,25 +106,26 @@ def start_single_worker(process_index: int, label: str, function: Callable) -> E
     return process
 
 
+# FIXME throw could possibly be removed from the signature.
+# This holds: throw == (worker_function is not None)
 def check_worker_exceptions(
     process_list: List[ExceptionSafeProcess],
     worker_label: str,
-    config: Optional[ConfigParser] = None,
+    throw: bool = True,
```

Comment:
I honestly fail to grasp what the parameter is supposed to do. Would you mind to explain?
I renamed it to `restart_workers`.

Reply:
The idea behind it was that you can either set it to "throw_exceptions" so that it would stop in case of an error and reraise or pass on the exception from inside the worker process so that is doesn't get lost ("dev/debugging mode") or to not "throw_exceptions" as a more robust mode for production which would restart the workers in case of an error so that the whole system could continue to run (but the naming was never very intuitive :sweat_smile:).
At least I think that was @dorpvom's intention.

Reply:
Thank you for the explanation.

Reply:
With that knowledge I changed the default value for testing to true.
This when a worker crashes pytest does not timeout but rather show the exception.